### PR TITLE
Move daemon-only flags into the daemon config struct

### PR DESCRIPTION
**Pull request by aidanhs**

### CODE REVIEW

**Comment by jfrazelle** [comment]
Can you please sign your commits following these rules:

https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

The easiest way to do this is to amend the last commit:

`git commit --amend -s --no-edit && git push -f`

**Comment by aidanhs** [comment]
Done and also fixed some gofmt issues.

The flag ordering seems to be completely arbitrary so I've just inserted them in a random place.

File: docker/flags.go [hunk 1]
```diff
@@ -37,15 +37,13 @@ func getDaemonConfDir() string {
...
-	flDebug       = flag.Bool([]string{"D", "-debug"}, false, "Enable debug mode")
-	flSocketGroup = flag.String([]string{"G", "-group"}, "docker", "Group to assign the unix socket specified by -H when running in daemon mode\nuse '' (the empty string) to disable setting of a group")
-	flLogLevel    = flag.String([]string{"l", "-log-level"}, "info", "Set the logging level (debug, info, warn, error, fatal)")
-	flEnableCors  = flag.Bool([]string{"#api-enable-cors", "-api-enable-cors"}, false, "Enable CORS headers in the remote API")
-	flTls         = flag.Bool([]string{"-tls"}, false, "Use TLS; implied by --tlsverify flag")
-	flHelp        = flag.Bool([]string{"h", "-help"}, false, "Print usage")
-	flTlsVerify   = flag.Bool([]string{"-tlsverify"}, dockerTlsVerify, "Use TLS and verify the remote (daemon: verify client, client: verify daemon)")
+	flVersion   = flag.Bool([]string{"v", "-version"}, false, "Print version information and quit")
+	flDaemon    = flag.Bool([]string{"d", "-daemon"}, false, "Enable daemon mode")
+	flDebug     = flag.Bool([]string{"D", "-debug"}, false, "Enable debug mode")
```
Comment (jfrazelle) [inline docker/flags.go]: No, these can be used elsewhere.

Comment (aidanhs) [inline docker/flags.go]: Where? The daemon is responsible for creating the socket and serving the API. These arguments don't apply to any other docker subcommands...

Comment (tianon) [inline docker/flags.go]: I know "docker info" at least has different behavior with "-D" passed to
_either_ the client or the daemon.

Comment (aidanhs) [inline docker/flags.go]: Which is exactly why I left "-D" alone.

Comment (aidanhs) [inline docker/flags.go]: (the only things changed are -G and --api-enable-cors)

Comment (tianon) [inline docker/flags.go]: :+1: (@jfrazelle triiicked me)

Comment (jfrazelle) [inline docker/flags.go]: Oh wow read this diff wrong

Comment (aidanhs) [inline docker/flags.go]: The tradeoff of enforced gofmt :)

**Comment by tiborvass** [comment]
@aidanhs Following your last comments on https://github.com/docker/docker/pull/10535#issuecomment-73141672

Docker maintainers (> Docker Inc), are currently overwhelmed by the 1.5 release hence our slow down in PR review. We read all comments, and we will review all PRs in due time, so please bear with us :)

Btw, your PR here seems very reasonable and we should be able to merge it reasonably quickly, so if you could rebase that'd be fantastic! Thanks for your patience!

**Comment by jfrazelle** [comment]
needs a rebase, so sorry, if you ping me after I will review immediately

**Comment by aidanhs** [comment]
@tiborvass that's fine, Docker is hardly a low activity project, I can imagine there are busy periods which is why I made my comment - no point spending time if maintainers are busy with other things :)

@jfrazelle ping

**Comment by tiborvass** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**